### PR TITLE
dr-autosync: add recover timeout

### PR DESCRIPTION
**Pull request by disksing**
### What problem does this PR solve?
Issue Number: Ref #4399

### What is changed and how does it work?
add a configuration item to delay dr state switch from async to sync-recover

```commit-message
```

### Check List

Tests
- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/replication/replication_mode.go [hunk 1]
```diff
@@ -477,7 +488,7 @@ func (m *ModeManager) tickUpdateState() {
 			m.drSwitchToAsync(storeIDs[primaryUp])
 		}
 	case drStateAsync:
-		if canSync {
+		if canSync && m.drDurationSinceAsyncStart() > m.config.DRAutoSync.WaitRecoverTimeout.Duration {
```
Comment (lhy1024) [inline pkg/replication/replication_mode.go:491]: Is this config only used to switch from `async` to `syncRecover`?

File: pkg/replication/replication_mode.go [hunk 2]
```diff
@@ -262,7 +263,8 @@ func (m *ModeManager) drSwitchToAsyncWithLock(availableStores []uint64) error {
 		log.Warn("failed to switch to async state", zap.String("replicate-mode", modeDRAutoSync), errs.ZapError(err))
 		return err
 	}
-	dr := drAutoSyncStatus{State: drStateAsync, StateID: id, AvailableStores: availableStores}
+	now := time.Now()
```
Comment (lhy1024) [inline pkg/replication/replication_mode.go:266]: Do we need to pick it to 6.5 and other branches?

**Review by lhy1024 — APPROVED** [review]

File: pkg/replication/replication_mode.go (hunk 1 above)
Comment (disksing) [reply pkg/replication/replication_mode.go:491]: yes!

File: pkg/replication/replication_mode.go (hunk 2 above)
Comment (disksing) [reply pkg/replication/replication_mode.go:266]: need cherry-pick 6.5. i have added the label

**Review by nolouch — APPROVED** [review]

**Pull request merged**